### ISSUE TITLE
util-linux & fontconfig: Wrong pkgconf might be picked up

### ISSUE THREAD

**Issue by Thyre** [easybuilders/easybuild-easyconfigs#21980]
On my personal system, where a lot of programs like `pkgconf`, `ccache` and so on are installed outside of EasyBuild, some packages fail to build with the existing EasyConfigs. With `utils-linux`, we run into the following issue:

Using `util-linux-2.40-GCCcore-13.3.0.eb`, an installation quickly aborts:
```
== configuring...
== ... (took 8 secs)
== FAILED: Installation ended unsuccessfully (build directory: /data/EasyBuild-develop/build/utillinux/2.40/GCCcore-13.3.0): build failed (first 300 chars): cmd " ./configure --prefix=/data/EasyBuild-develop/software/util-linux/2.40-GCCcore-13.3.0  --build=x86_64-pc-linux-gnu  --host=x86_64-pc-linux-gnu 
--disable-chfn-chsh --disable-login --disable-su --disable-rfkill --disable-wall --disable-use-tty-group --disable-makeinstall-chown --disable-makeinsta (took 9 secs)
== Results of the build can be found in the log file(s) /tmp/eb-k84hfnfs/easybuild-util-linux-2.40-20241206.095048.VAjjd.log
```

The log doesn't really help here, but `config.log` gives some hints:

```
1112 configure:22005: checking for x86_64-pc-linux-gnu-pkg-config
1113 configure:22028: found /usr/bin/x86_64-pc-linux-gnu-pkg-config
1114 configure:22040: result: /usr/bin/x86_64-pc-linux-gnu-pkg-config
1115 configure:22113: checking pkg-config is at least version 0.9.0
[...]
8487 configure:31294: checking for sqlite3
8488 configure:31353: result: no 
8489 configure:31382: error: liblastlog2 selected, but required sqlite3 library not available
```

The incorrectly chosen `pkgconf` causes issues when checking for other libraries, causing `configure` to abort. In a quick test, I was able to work around this with:

```diff
diff --git a/easybuild/easyconfigs/u/util-linux/util-linux-2.40-GCCcore-13.3.0.eb b/easybuild/easyconfigs/u/util-linux/util-linux-2.40-GCCcore-13.3.0.eb
index 7ca00a929a..8449c25d18 100644
--- a/easybuild/easyconfigs/u/util-linux/util-linux-2.40-GCCcore-13.3.0.eb
+++ b/easybuild/easyconfigs/u/util-linux/util-linux-2.40-GCCcore-13.3.0.eb
@@ -37,6 +37,8 @@ configopts += "--with-systemdsystemunitdir='${prefix}/systemd' "
 configopts += "--with-bashcompletiondir='${prefix}/share/bash-completion/completions' "
 # disable building Python bindings (since we don't include Python as a dep)
 configopts += "--without-python "
+configopts += "--with-pkgconfigdir=$EBROOTPKGCONF "
+configopts += "PKG_CONFIG=$EBROOTPKGCONF/bin/pkgconf "
 
 sanity_check_paths = {
     'files': ['lib/lib%s.a' % x for x in ['blkid', 'mount', 'uuid']],
```

Not sure if `--with-pkgconfigdir` is needed. Setting both, the installation succeeded.

**Comment by Thyre** [comment]
The same thing happens with `fontconfig`:

```
291 configure:5516: checking for x86_64-pc-linux-gnu-pkg-config
292 configure:5539: found /usr/bin/x86_64-pc-linux-gnu-pkg-config
293 configure:5551: result: /usr/bin/x86_64-pc-linux-gnu-pkg-config
294 configure:5624: checking pkg-config is at least version 0.9.0
295 configure:5630: result: no 
[...]
2170 configure:18745: checking for FREETYPE
2171 configure:18813: result: no
2172 configure:18815: error: in `/data/EasyBuild-develop/build/fontconfig/2.15.0/GCCcore-13.3.0/fontconfig-2.15.0':
2173 configure:18817: error: The pkg-config script could not be found or is too old.  Make sure it
2174 is in your PATH or set the PKG_CONFIG environment variable to the full
2175 path to pkg-config.
2176 
2177 Alternatively, you may set the environment variables FREETYPE_CFLAGS
2178 and FREETYPE_LIBS to avoid the need to call pkg-config.
2179 See the pkg-config man page for more details.
2180 
2181 To get pkg-config, see <http://pkg-config.freedesktop.org/>.
2182 See `config.log' for more details
2183 
2184 ## ---------------- ##
```

**Comment by Thyre** [comment]
While searching through our PRs, I found some which set `PKG_CONFIG` in the EasyConfig.
The question is: Can we set this as an environment variable (i.e. does it get picked up correctly by `configure`) and what would be the preferred solution?

**Comment by Micket** [comment]
Well it's really only done in ImageMagick, where apparently, it needs the extra hint as to where to find pkg-config (despite the fact that we do of course put it into PATH like they request here, it's just these configure tools that try to be clever and look for the x86_64-xxx-linux-gnu-pkg-config variants first)

I see no reason why `pkgconf` easyconfig could just set `PKG_CONFIG` variable globally whenever it's loaded. It would solve this problem, and I wouldn't expect it to have any negative side effects since you absolutely don't want to mix in any OS pkgconf once this module is loaded.

**Comment by Micket** [comment]
I think this is fixed. If not please reopen

**Comment by Thyre** [comment]
Unfortunately, this issue is not entirely solved.

I ran into this again with current `develop` and a re-installed `pkgconf` when trying to build `wxWidgets-3.2.6-GCC-13.3.0.eb`.
Even trying into a container for my OS (Arch Linux), I ran into the same failure as `pkg-config` was installed by some other system package. In this case, I had to run EasyBuild with `PKG_CONFIG=$EBROOTPKGCONF/bin/pkg-config` for building to succeed.
